app/models/discussion.rb

```diff
@@ -4,14 +4,48 @@ class Discussion < ActiveRecord::Base
              foreign_key: 'author_id',
              required: true
 
+  has_many :comments, dependent: :destroy
+
   validates_presence_of :fid
 
-  scope :about_fid, ->(fid) { where(fid: fid).order(:created_at) }
+  before_destroy :check_comments, prepend: true
```

Comment:
Now that took me a while. prepend: true is there so the destroy cascade is not run before. This is quite counter-intuitive.